packages/react-components/react-list-preview/src/components/List/List.types.ts

```diff
@@ -1,17 +1,34 @@
-import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import * as React from 'react';
+import type { ComponentProps, ComponentState, Slot, SelectionMode, SelectionItemId } from '@fluentui/react-utilities';
+import type { ListSelectionState } from '../../hooks/types';
+import { ListAccessibilityRoles } from '../../hooks/useListAccessibilityRoles';
 
 export type ListSlots = {
-  root: Slot<'div'>;
+  root: NonNullable<Slot<'ul', 'div' | 'ol'>>;
 };
 
 /**
  * List Props
  */
-export type ListProps = ComponentProps<ListSlots> & {};
+export type ListProps = ComponentProps<ListSlots> & {
+  selectionMode?: SelectionMode;
+  selectedItems?: SelectionItemId[];
+  defaultSelectedItems?: SelectionItemId[];
+  // eslint-disable-next-line @nx/workspace-consistent-callback-type -- can't change type of existing callback
```

Comment:
This is a new component - there's no need to inherit past debt. You should be able to properly type this callback